code/game/machinery/vending.dm

```diff
@@ -949,6 +968,38 @@ var/global/num_vending_terminals = 1
 			new_price = R.price
 		R.price = new_price
 
+	else if (href_list["delete_entry"] && src.vend_ready && !currently_vending)
+		//testing("vend: [href]")
+
+		if (!allowed(usr) && !emagged && scan_id) //For SECURE VENDING MACHINES YEAH
+			to_chat(usr, "<span class='warning'>Access denied.</span>")//Unless emagged of course
+
+			flick(src.icon_deny,src)
+			return
+
+		var/idx=text2num(href_list["delete_entry"])
+		var/cat=text2num(href_list["cat"])
+
+		var/datum/data/vending_product/R = GetProductByID(idx,cat)
+		if (!R || !istype(R) || !R.product_path)
+			message_admins("Invalid vend request by [formatJumpTo(src.loc)]: [href]")
```

Comment:
now I wonder here what happens if someone puts naughty things in their href and it gets broadcasted to every admin online :thinking:

Reply:
The href variable is dictated by which button is being pressed in the interface. It isn't user-specified so there are only 8 or so possible values for href here. But I do see where it could possibly be spammed to piss off admins.

Reply:
It doesn't really matter now that the message was removed but for future reference, everything in hrefs can be "hacked" by the clients. They can send values other than the default buttons, including raw HTML, which is why everything regarding hrefs has to be obsessively sanitized